aggregator/pair.go

```diff
@@ -0,0 +1,71 @@
+package aggregator
+
+import (
+	"fmt"
+	"math"
+)
+
+const (
+	minPercentDifferenceToNotify = 1
+	minDecimals                  = 1
+	maxDecimals                  = 18
+)
+
+// ArgsPair is the argument DTO for a pair
+type ArgsPair struct {
+	Base                      string
+	Quote                     string
+	PercentDifferenceToNotify uint32
+	Decimals                  uint64
+	Exchanges                 map[string]struct{}
+}
+
+type pair struct {
```

Comment:
this started to look as a hybrid: DTO-like struct with functionality (sort-of)
If we stop here with the functionality, we can say we are safe.
What I do not like is the fact that I can create an instance using the `NewPair` function using some arguments that will pass all checks but then I can easily alter them in an unsafe way. Let's say, I can do something like:
```GO
p := NewPair(args)
p.TrimPrecision = 0
```
which can cause panics if we need to divide somewhere with the TrimPrecision value

This happens because the apparent safety provided by the `NewPair` constructor that can signal errors.

Reply:
done. all priorities and constructor are now unexported